pkg/client/vault/client.go

```diff
@@ -205,3 +304,95 @@ func compressZCAP(zcap *zcapld.Capability) (string, error) {
 
 	return base64.URLEncoding.EncodeToString(compressed.Bytes()), nil
 }
+
+func lastElm(s, sep string) string {
+	all := strings.Split(s, sep)
+
+	return all[len(all)-1]
+}
+
+func buildEDVURI(h, vid, did string) string {
+	return fmt.Sprintf("%s/encrypted-data-vaults/%s/documents/%s", h, vid, did)
+}
+
+func prepareRemoteJWEKey(keyURL string, kmsImpl kms.KeyManager) ([]byte, string, error) {
+	kid := lastElm(keyURL, "/")
+
+	pubKeyBytes, err := kmsImpl.ExportPubKeyBytes(kid)
+	if err != nil {
+		return nil, "", fmt.Errorf("failed to retrieve main public key bytes from remote KMS: %w", err)
+	}
+
+	return pubKeyBytes, kid, nil
+}
+
+func encryptKey(wKMS KeyManager, wCrypto crypto.Crypto, cek []byte) (string, error) {
+	_, kidURL, err := wKMS.Create(kms.NISTP256ECDHKW)
+	if err != nil {
+		return "", fmt.Errorf("create: %w", err)
+	}
+
+	pubKeyBytes, jweCryptoKID, err := prepareRemoteJWEKey(kidURL.(string), wKMS)
+	if err != nil {
+		return "", fmt.Errorf("prepare remote JWE key: %w", err)
+	}
+
+	ecPubKey := &crypto.PublicKey{KID: jweCryptoKID}
+
+	err = json.Unmarshal(pubKeyBytes, ecPubKey)
+	if err != nil {
+		return "", fmt.Errorf("unmarshal: %w", err)
+	}
+
+	encrypter, err := jose.NewJWEEncrypt(jose.A256GCM, jose.DIDCommEncType, "", nil,
```

Comment:
`encType` should be "A256GCM" in this case:

* we are using a 32-byte key
* we are using AESGCM to encrypt

Reply:
You need to provide the `senderKID` and `senderKH` parameters because you are encrypting to your own key